### PR TITLE
Add adr for using js wrapper when embedding iframe

### PR DESCRIPTION
**Pull request by code28**

### CODE REVIEW

**Review by pgurusinga — APPROVED** [review]

File: doc/adr/0004-embedding-widget-via-js-wrapper.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+The eService (Service Provider) embeds a widget displaying a QR code, which will be scanned on a mobile device. After finishing the eID process on the mobile device, the user needs to be redirected to the `refreshAddress`, which the eID client app received from the TC token.
+
+The refresh address is sent via a message proxy server from the eID client app to the widget page.
+
+Iframes can't update the parent's `location.href` due to the same-origin policy, so redirecting the user out of the iframe is not possible.
+
+## Decision
+
+We provide a JavaScript wrapper file for embedding the widget iframe. By embedding a JS file, we can update the `location.href`. To communicate between widget and wrapper JS, we use cross-document messaging (`postMessage` and `window.onmessage`).
```
Comment (mpanne) [inline doc/adr/0004-embedding-widget-via-js-wrapper.md:19]: Maybe rephrase to make clear what cross-document messaging is used for? Just a thought. 

"We provide a JavaScript wrapper file which will be included in the eService page and which creates the widget iframe. This allows us to communicate between the widget and the parent page via cross-document messaging (`postMessage` and `window.onmessage`) and therefore update the `location.href`."

**Review by mpanne — APPROVED** [review]

File: doc/adr/0004-embedding-widget-via-js-wrapper.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Accepted
+
+## Context
+
+The eService (Service Provider) embeds a widget displaying a QR code, which will be scanned on a mobile device. After finishing the eID process on the mobile device, the user needs to be redirected to the `refreshAddress`, which the eID client app received from the TC token.
+
+The refresh address is sent via a message proxy server from the eID client app to the widget page.
+
+Iframes can't update the parent's `location.href` due to the same-origin policy, so redirecting the user out of the iframe is not possible.
```
[the remark this answers is not in the document]
Comment (code28) [reply doc/adr/0004-embedding-widget-via-js-wrapper.md:15]: Oh, yes, good point!

File: doc/adr/0004-embedding-widget-via-js-wrapper.md (hunk 1 above)
Comment (code28) [reply doc/adr/0004-embedding-widget-via-js-wrapper.md:19]: Love it. 👍
